include/ada/url.h

```diff
@@ -368,6 +368,13 @@ namespace ada {
      * @see https://github.com/servo/rust-url/blob/b65a45515c10713f6d212e6726719a020203cc98/url/src/quirks.rs#L31
      */
     [[nodiscard]] ada_really_inline ada::url_components get_components() noexcept;
+    
+    /**
+     * Set the specified value as the fragment of this URL.
+     * The string is not processed, this is a direct assignment.
```

Comment:
I think we should always process it. I don't think of a scenario where Node.js or any other runtime does not need any processing of the input.

Whether or not we need processing is an optimization we can do to not-allocate additional resources than we need.